crates/re_viewer/src/app.rs

```diff
@@ -833,75 +829,51 @@ impl eframe::App for App {
             render_ctx.gpu_resources.statistics()
         };
 
-        // Look up the blueprint in use for this frame.
-        // Note that it's important that we save this because it's possible that it will be changed
-        // by the end up the frame (e.g. if the user selects a different recording), but we need it
-        // to save changes back to the correct blueprint at the end.
-        let active_blueprint_id = self
-            .state
-            .selected_blueprint_by_app
-            .get(&self.selected_app_id())
-            .cloned()
-            .unwrap_or_else(|| {
-                StoreId::from_string(StoreKind::Blueprint, self.selected_app_id().0)
-            });
-
-        let store_stats = self
-            .recording_db()
-            .map(|store_db| DataStoreStats::from_store(&store_db.entity_db.data_store))
-            .unwrap_or_default();
-
-        let blueprint_stats = self
-            .store_hub
-            .blueprint_mut(&active_blueprint_id)
-            .map(|bp_db| DataStoreStats::from_store(&bp_db.entity_db.data_store))
-            .unwrap_or_default();
+        // TODO(jleibs): Work through this ordering. Would be great to move the
+        // memory panel update *after* the calls to purge/receive so we don't
+        // need to generate the view twice.
+        let store_stats = store_hub.stats();
```

Comment:
not quite following. What does it mean to generate the view twice?

Reply:
Ahh, good call. Obsoleted comment based on moving stats from accessor on the StoreContext to directly off the store.